.github/workflows/tools.yml

```diff
@@ -0,0 +1,129 @@
+name: Build host tools
+
+on:
+  pull_request:
+    paths:
+      - 'tools/**'
+
+jobs:
+  build:
+    name: Build tools on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: False
+      matrix:
+        os: 
+          - ubuntu-latest
+          - macos-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          path: openwrt
+
+      - name: Setup MacOS
+        if: ${{ matrix.os == 'macos-latest' }}
+        run: |
+          echo "WORKPATH=/Volumes/OpenWrt/openwrt/" >> "$GITHUB_ENV"
+          hdiutil create -size 20g -type SPARSE -fs "Case-sensitive HFS+" -volname OpenWrt OpenWrt.sparseimage
+          hdiutil attach OpenWrt.sparseimage
+          mv "$GITHUB_WORKSPACE/openwrt" /Volumes/OpenWrt/
+          cd "$WORKPATH"
+
+          brew install \
+            autoconf \
+            automake \
+            coreutils \
+            diffutils \
+            findutils \
+            gawk \
+            gettext \
+            git-extras \
+            gmp \
+            gnu-getopt \
+            gnu-sed \
+            gnu-tar \
+            grep \
+            libidn2 \
+            libunistring \
+            m4 \
+            make \
+            mpfr \
+            ncurses \
+            openssl@1.1 \
+            pcre \
+            pkg-config \
+            quilt \
+            readline \
+            wget \
+            zstd
+
+            echo "/bin" >> "$GITHUB_PATH"
+            echo "/sbin/Library/Apple/usr/bin" >> "$GITHUB_PATH"
+            echo "/usr/bin" >> "$GITHUB_PATH"
+            echo "/usr/local/bin" >> "$GITHUB_PATH"
+            echo "/usr/local/opt/coreutils/bin" >> "$GITHUB_PATH"
+            echo "/usr/local/opt/findutils/libexec/gnubin" >> "$GITHUB_PATH"
+            echo "/usr/local/opt/gettext/bin" >> "$GITHUB_PATH"
+            echo "/usr/local/opt/gnu-getopt/bin" >> "$GITHUB_PATH"
+            echo "/usr/local/opt/make/libexec/gnubin" >> "$GITHUB_PATH"
+            echo "/usr/local/opt/make/libexec/gnubin" >> "$GITHUB_PATH"
+            echo "/usr/sbin" >> "$GITHUB_PATH"
+            pwd
+
+      - name: Setup Ubuntu
```

Comment:
FYI using latest Ubuntu can produce misleading CI results as it might then fail on buildbots which are based on [Debian 10](https://gitlab.com/openwrt/buildbot/-/blob/master/docker/buildworker/Dockerfile), so likely different versions of everything. So you should probably consider either using that buildworker container in this job (as a bonus you don't need to waste more CO2/resources/bandwidth with this apt-get foo on every push in the PR) or add another job which is going to assure, that it probably won't bomb out on buildbots.

Reply:
I'll look into that. It's not really possible for macOS sadly. Since we offer setup guides for both macOS as well as Linux (Debian/Ubuntu/Alpine/NixOS) I thought this non-dockerized build might show build failures more easily.

Do you locally compile within a container? I've been looking at solutions like [cake](https://github.com/wurosh/cake) to minimize build system issues.

Reply:
> Do you locally compile within a container?

Nope, I just build test before pushing via https://gitlab.com/ynezz/openwrt-ci/-/blob/master/openwrt-ci/gitlab/target.yml